File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int vis[200];
vector<vector<int>> computers;
void go(int node){
    for(int i=0;i<computers.size();i++){
        if(!vis[i]&&computers[node][i]){
            vis[i]=1;
            go(i);
        }
    }
}
int solution(int n, vector<vector<int>> computers) {
    ::computers=computers;
    int answer = 0;
    for(int i=0;i<n;i++){
        if(!vis[i]){
            vis[i]=1;
            go(i);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    
    int[][] computers;
    int[] vis = new int[200];
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        this.computers = computers;
        for (int i = 0; i < n; i++) {
            if (vis[i] == 0) {
                vis[i] = 1;
                go(i);
                answer++;
            }
        }
        return answer;
    }
    
    void go(int node) {
        for (int i = 0; i < computers.length; i++) {
            if (vis[i] == 0 && computers[node][i] > 0) {
                vis[i] = 1;
                go(i);
            }
        }
    }
}